WDE.SmartScriptEditor/Providers/SmartDataPresentersProvider/SmartDataEditorsProvider.cs

```diff
@@ -24,21 +25,21 @@ class SmartDataEditorsProvider : IMainMenuItem
         public MainMenuItemSortPriority SortPriority { get; } = MainMenuItemSortPriority.PriorityNormal;
 
         public SmartDataEditorsProvider(ISmartRawDataProvider smartDataProvider, IParameterFactory parameterFactory, ISmartDataManager smartDataManager,
-            ITaskRunner taskRunner, IMessageBoxService messageBoxService, IWindowManager windowManager)
+            ITaskRunner taskRunner, IMessageBoxService messageBoxService, IWindowManager windowManager, IHistoryManager history)
```

Comment:
Each instance of those documents needs a new instance of IHistoryManager. If you pass `history` that was provided via constructor of SmartDataEditorsProvider, then each document will have the same HistoryManager.

In order to achieve this (get multiple instances of same class in the single object), you need to take `Func<IHistoryManager>` in `SmartDataEditorsProvider` constructor. With this, each time you invoke that function, you will get a new instance